apps/guardian-ui/src/components/SetConfiguration.tsx

```diff
@@ -71,10 +77,12 @@ export const SetConfiguration: React.FC<Props> = ({ next }: Props) => {
     });
   const [mintAmounts, setMintAmounts] = useState<number[]>([]);
   const [error, setError] = useState<string>();
+  const [otherModuleParams, setOtherModuleParams] = useState<object>({});
 
   useEffect(() => {
     const initStateFromParams = (params: ConfigGenParams) => {
       setFederationName(params.meta?.federation_name || '');
+      setOtherModuleParams(getOtherModuleParamsFromConfig(params));
```

Comment:
IMO this is not necessary to filter out the old modules and makes this unnecessarily complicate. Just store `params`, and then we can call `submitConfiguration` with:
```ts
submitConfiguration({
  modules: {
    ...params,
    // all the known 1st party modules after the spread
  },
);
```